From 15695dc68befbee509df9eb6a7dcbac12e50613c Mon Sep 17 00:00:00 2001
From: "Holesch, Simon (BSH)" <simon.holesch@bshg.com>
Date: Mon, 11 Feb 2019 09:47:46 +0100
Subject: [PATCH] udhcp: Add proprietary BSH DHCP Options

DHCP servers in the factory are configured to include two DHCP
options:
- BSH_SMM_OIS: Used for the deprecated pull process, where the SMM
  actively connects to the OIS and requests the firmware to install.
- BSH_SMM_Registry: Used for the newer push process, where the SMM
  registers itself to the given URL and waits passively for instructions
  over SSH.

---
 networking/udhcp/common.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/networking/udhcp/common.c b/networking/udhcp/common.c
index 680852c..16691bc 100644
--- a/networking/udhcp/common.c
+++ b/networking/udhcp/common.c
@@ -65,6 +65,8 @@ const struct dhcp_optflag dhcp_optflags[] = {
 	{ OPTION_STRING                           , 0xd1 }, /* DHCP_PXE_CONF_FILE */
 	{ OPTION_STRING                           , 0xd2 }, /* DHCP_PXE_PATH_PREFIX */
 	{ OPTION_6RD                              , 0xd4 }, /* DHCP_6RD           */
+	{ OPTION_STRING               | OPTION_REQ, 0xe0 }, /* BSH OIS            */
+	{ OPTION_STRING               | OPTION_REQ, 0xe1 }, /* BSH SMM Registry   */
 	{ OPTION_STATIC_ROUTES | OPTION_LIST      , 0xf9 }, /* DHCP_MS_STATIC_ROUTES */
 	{ OPTION_STRING                           , 0xfc }, /* DHCP_WPAD          */
 
@@ -133,6 +135,8 @@ const char dhcp_option_strings[] ALIGN1 =
 	"pxeconffile" "\0" /* DHCP_PXE_CONF_FILE  */
 	"pxepathprefix" "\0" /* DHCP_PXE_PATH_PREFIX  */
 	"ip6rd" "\0"       /* DHCP_6RD            */
+	"bshois" "\0"
+	"bshsmmregistry" "\0"
 	"msstaticroutes""\0"/* DHCP_MS_STATIC_ROUTES */
 	"wpad" "\0"        /* DHCP_WPAD           */
 	;
-- 
2.7.4

